hcache/hcache.c

```diff
@@ -437,11 +555,27 @@ int mutt_hcache_store_raw(header_cache_t *hc, const char *key, size_t keylen,
     return -1;
 
   struct Buffer path = mutt_buffer_make(1024);
+  int rc;
 
   keylen = mutt_buffer_printf(&path, "%s%s", hc->folder, key);
 
-  int rc = ops->store(hc->ctx, mutt_b2s(&path), keylen, data, dlen);
+#ifdef USE_HCACHE_COMPRESSION
+  if (C_HeaderCacheCompressMethod)
+  {
+    const struct ComprOps *cops = compr_get_ops();
+
+    size_t clen;
+    void *cbuf = cops->compress(hc->cctx, data, dlen, &clen);
+    rc = ops->store(hc->ctx, mutt_b2s(&path), keylen, cbuf, clen);
+    mutt_buffer_dealloc(&path);
+
+    return rc;
```

Comment:
Please remove the duplication of store / dealloc / return. You could override `data` inside the `#ifdef` and simpy flow outside the `ifdef`. Also, does `cbuf` need deallocating after the store?

Reply:
Oh, I see `compress` always return a buffer owned by the context. If this is a requirement (i.e., callers are not supposed to free it, because it's going to be free'd on close) perhaps it makese sense to document it in the API.

Reply:
I still suggest to simplify the conditional so just properly set `data` and `dlen`.

```c
if (C_HeaderCacheCompressMethod)
{
  const struct ComprOps *cops = compr_get_ops();
  data = cops->compress(hc->cctx, data, dlen, &dlen);
}
```